test/models/TraderProfileSpec.scala

```diff
@@ -376,4 +377,124 @@ class TraderProfileSpec extends AnyFreeSpec with Matchers with TryValues with Op
       }
     }
   }
+
+  ".validateHasNirms" - {
+
+    "must validate Nirms" - {
+
+      "user has Nirms and changes answer to No" in {
+        val userProfile = TraderProfile(testEori, "1", Some("nirms"), None)
+
+        val answers =
+          UserAnswers(userAnswersId)
+            .set(HasNirmsUpdatePage, false)
+            .success
+            .value
+            .set(RemoveNirmsPage, true)
+            .success
+            .value
+            .set(TraderProfileQuery, userProfile)
+            .success
+            .value
+
+        val result = TraderProfile.validateHasNirms(answers)
+
+        result mustEqual Right(false)
+      }
+
+      "user does not have Nirms and does not change answer to Yes" in {
+        val userProfile = TraderProfile(testEori, "1", None, None)
+
+        val answers =
+          UserAnswers(userAnswersId)
+            .set(HasNirmsUpdatePage, false)
+            .success
```

Comment:
Is removePage needed here at all?

Reply:
I think no. This is why we need to add TraderProfileQuery (to skip the warning page)

Scenario: 
1. User does not have Nirms 
2. Click change link
3. Do not change answer (Select No)
4. Should go to CYA (not to warning page)